### ISSUE TITLE
upgrading EKS cluster hosting Kafka

### ISSUE THREAD

**Issue by hjoshi84** [aws-samples/eks-cluster-upgrade#107]
### Use case

I am trying to update the EKS cluster hosting Kafka. The script upgraded nodes one by one. Is there any provision where the script can wait some specified time until the next node upgrade or waits till the Kafka partition syncs up?

### Solution/User Experience

None

**Comment by quixoticmonk** [comment]
To clarify, are you asking for a wait state between node upgrade based on user input ?

**Comment by bryantbiggs** [comment]
I don't know if this should be handled here - I think this is better handled through the normal Kubernetes highly available constructs such as startup probes, readiness probes, PDBs, etc. 

How are you deploying Kafka - are you using the Strimzi operator?

**Comment by hjoshi84** [comment]
> I don't know if this should be handled here - I think this is better handled through the normal Kubernetes highly available constructs such as startup probes, readiness probes, PDBs, etc.
> 
> How are you deploying Kafka - are you using the Strimzi operator?

I am using confluent kafka as operator

**Comment by hjoshi84** [comment]
> 

Yes. before upgrading the next node script should wait until Kafka pods sync the partitions. otherwise, there will be a loss of data. I believe this will be a good feature addition.

**Comment by bryantbiggs** [comment]
and how would one check if a kafka pod has finished syncing? isn't this what statefulsets in combination with pod disruption budgets are intended to handle?
